### PR TITLE
feat(cloud): add strapi cloud plugin to CMS for marketing

### PR DESCRIPTION
**Pull request by Mcastres**
### What does it do?

This PR adds a Strapi plugin Cloud displaying, in the short term, instructions to deploy your project on Strapi Cloud.
(Versioning on GitHub > Deploy on Strapi Cloud)

### Why is it needed?

Give more visibility to Strapi Cloud within the product in the short term. Create a bridge between our two product in the long term through this plugin

### Questions

I'm not sure if:
- The way I check if the project is versioned on git via the simple presence of the `.git` folder is enough. Note that this is only for having a specific UI based on this but this is not mandatory for the plugin.
- This is the right place to push this plugin.

### CODE REVIEW

**Comment by derrickmehaffy** [comment]
> The way I check if the project is versioned on git via the simple presence of the .git

FYI we automatically init git when a project is created, not sure if it impacts your decision here but just wanted to make that known.

**Comment by derrickmehaffy** [comment]
While I assume this is implied but can you confirm the following:

- will this be installed by default for new projects?
- can it be removed?
- what about users who are already using cloud, should it still be shown?

File: packages/plugins/cloud/README.md
Comment (joshuaellis) [inline packages/plugins/cloud/README.md:1]: Needs to be filled in.

File: packages/plugins/cloud/admin/src/components/Cloud/index.js
Comment (joshuaellis) [inline packages/plugins/cloud/admin/src/components/Cloud/index.js:8]: rogue comment

File: packages/plugins/cloud/admin/src/components/Cloud/index.js
Comment (joshuaellis) [inline packages/plugins/cloud/admin/src/components/Cloud/index.js:31]: Why does this need a key?

File: packages/plugins/cloud/admin/src/components/Cloud/index.js
Comment (joshuaellis) [inline packages/plugins/cloud/admin/src/components/Cloud/index.js:6]: wrong name

File: packages/plugins/cloud/admin/src/components/Icons/check.js
Comment (joshuaellis) [inline packages/plugins/cloud/admin/src/components/Icons/check.js:1]: Why isn't this using the design-system's icons?

File: packages/plugins/cloud/admin/src/components/Icons/link.js
Comment (joshuaellis) [inline packages/plugins/cloud/admin/src/components/Icons/link.js:1]: same question

File: packages/plugins/cloud/admin/src/components/Icons/upload.js
Comment (joshuaellis) [inline packages/plugins/cloud/admin/src/components/Icons/upload.js:1]: same question

File: packages/plugins/cloud/admin/src/index.js
Comment (joshuaellis) [inline packages/plugins/cloud/admin/src/index.js:29]: if you've commented this, I assume it's not needed

File: packages/plugins/cloud/admin/src/index.js
Comment (joshuaellis) [inline packages/plugins/cloud/admin/src/index.js:19]: you've not changed the chunkName.

File: packages/plugins/cloud/admin/src/pages/HomePage/index.js
Comment (joshuaellis) [inline packages/plugins/cloud/admin/src/pages/HomePage/index.js:75]: This is a bad way to fetch data. You should use react-query like we do in the rest of the codebase please.

File: packages/plugins/cloud/admin/src/utils/api.js
Comment (joshuaellis) [inline packages/plugins/cloud/admin/src/utils/api.js:14]: I would inline this in the component and it should use the `useFetchClient` hook from `helper-plugin`, not request.

File: packages/plugins/cloud/package.json
Comment (joshuaellis) [inline packages/plugins/cloud/package.json:36]: This should be filled out like other plugins or removed.

**Review by joshuaellis — CHANGES_REQUESTED** [review]
I think new features / packages are supposed to be written in typescript because this adds additional workload to the engineers to refactor. Also there's no eslint config so that needs to be added and the code formatted before we introduce this into the codebase.

RE. checking the git information, it'd be good to know if what resources you've already checked out before we proceed – i'm not sure checking if the folder exists works well, this might cause it to show in prod environments for example?

Does this plugin need to be in the monorepo? If so why & we should add contributor documentation even if it's one page – to explain the package clearly and for example how we're fetching the git info and why etc.

**Comment by Mcastres** [comment]
> // Uncomment to set the permissions of the plugin here
>         // {
>         //   action: '', // the action name should be plugin::plugin-name.actionType
>         //   subject: null,
>         // },

To be fully honest with you, I started to create it using TypeScript and the plugin was unable to find my controller method saying it was undefined even tho the code was absolutely identical (ts version of course) to this one. I tried to find the issue without success, maybe this is a bug with ts tho... 

I'm not pushing the final proposal here, I should have submitted a draft PR... 
Not sure this is the best solution concerning the `.git` folder. Indeed, it might appear on production env. Is it possible to prevent a plugin based on the env you are running? If yes that would be great.

> Does this plugin need to be in the mono repo? 

I don't know to be honest but can it be outside and installed by default?

**Comment by joshuaellis** [comment]
> To be fully honest with you, I started to create it using TypeScript and the plugin was unable to find my controller method saying it was undefined even tho the code was absolutely identical (ts version of course) to this one. 

You could raise this in slack with the DX squad for some assistance :)

>  Is it possible to prevent a plugin based on the env you are running?

Not really.

> can it be outside and installed by default

No.

**Comment by Mcastres** [comment]
> > The way I check if the project is versioned on git via the simple presence of the .git
> 
> FYI we automatically init git when a project is created, not sure if it impacts your decision here but just wanted to make that known.

Should I check for a specific file in the folder then?

**Comment by Mcastres** [comment]
> While I assume this is implied but can you confirm the following:
> 
> * will this be installed by default for new projects?

That would be the idea :)

> * can it be removed?

I hope so but I am not familiar enough to be sure how you can remove it. This PR is there to discuss that 😅 

> * what about users who are already using the cloud, should it still be shown?

I don't think there is a way for us to know within the product that the project is hosted on the cloud right? @joshuaellis said that it was not possible to "disable" a plugin based on the env you are running...

> RE. checking the git information, it'd be good to know if what resources you've already checked out before we proceed – i'm not sure checking if the folder exists works well, this might cause it to show in prod environments for example?

That's the first idea I came up with. Indeed, since we git init during project creation, maybe I should check for a specific file in this folder that is there only after pushing once? A reminder that this is not mandatory for the plugin, just a better UI presentation but this can be removed.

> Does this plugin need to be in the monorepo?

If it cannot be installed by default without being inside of it, yes I guess then

**Comment by joshuaellis** [comment]
> I hope so but I am not familiar enough to be sure how you can remove it. This PR is there to discuss that 😅

You're not going to be able to backport this to old users AND make it "un-installable" so which do you prefer? the choice impacts how we include this. e.g if we want it uninstallable we'll have to treat it like we do `user-permissions` and `i18n` 👍🏼 

> I don't think there is a way for us to know within the product that the project is hosted on the cloud right? @joshuaellis said that it was not possible to "disable" a plugin based on the env you are running...

I don't believe you can disable plugins based on conditions no, but you could look at the `window.strapi` object and never show UI based on conditions? IIRC we do know if the project is in the cloud.

> That's the first idea I came up with. Indeed, since we git init during project creation, maybe I should check for a specific file in this folder that is there only after pushing once? A reminder that this is not mandatory for the plugin, just a better UI presentation but this can be removed.

It might be worth talking to devops and seeing if they have an idea of how you can identify if something is on git. Otherwise, i'd talk to people like Ben or JS.

**Comment by joshuaellis** [comment]
> After discussing with Alex, we'll no longer check if the project has been versioned on git. I added the following:

so you can remove the server package in that case? :)

> Although I would need some indications on how to update the generator code to include this plugin by default 

Yeah you need to add it to the template so the templates are written correctly, you'll also probably need to update `generateTestApp` & the build script of the admin panel too.

When you're ready for a code review just re-request one 👍🏼

**Comment by Mcastres** [comment]
I think its ready now :D

**Comment by joshuaellis** [comment]
@Mcastres do you know Typescript? because someone in the next few weeks is going to have to convert this to TS and it will probably be me – i'd really appreciate it if you could convert it to TS to be honest, if you don't know it then just let me know and we can carry on.

**Comment by Mcastres** [comment]
Hey @joshuaellis, you can find the errors I'm facing:

```
[ERROR] admin/src/components/Cloud.tsx:17:31 - TS2615: Type of property 'defaultProps' circularly references itself in mapped type '{ readonly $$typeof: symbol; defaultProps?: Partial<WithOptionalTheme<MakeAttrsOptional<StyledComponentInnerComponent<WithC>, any, any, ComponentPropsWithRef<StyledComponentInnerComponent<WithC> extends ComponentType<...> | keyof IntrinsicElements ? (ComponentType<...> | keyof IntrinsicElements) & StyledComponentInn...'.
```

This is happening in the `admin/src/components/Cloud.tsx` file although there is the exact same code in `admin/src/components/Github.tsx`...

---

```
[ERROR] admin/src/pages/HomePage.tsx:15:32 - TS2307: Cannot find module './assets/corner-ornament.svg' or its corresponding type declarations.
[ERROR] admin/src/pages/HomePage.tsx:16:31 - TS2307: Cannot find module './assets/left-side-cloud.png' or its corresponding type declarations.
[ERROR] admin/src/pages/HomePage.tsx:17:32 - TS2307: Cannot find module './assets/right-side-cloud.png' or its corresponding type declarations.
```

The web indicates to add the following in the `custom.d.ts` file:

```
declare module '*.svg';
declare module '*.png';
```

But it doesn't work.

--- 

```
[ERROR] admin/src/pages/HomePage.tsx:106:14 - TS2322: Type '{ children: string; href: string; }' is not assignable to type 'IntrinsicAttributes & (LinkProps & RefAttributes<HTMLAnchorElement>)'.
  Property 'to' is missing in type '{ children: string; href: string; }' but require
```

**Comment by joshuaellis** [comment]
Your `declare module` issue, i fixed by separating into two files as I noticed the global & module worked independently but not togather. The `styled-components` stuff, after a quick google I found [this issue](https://github.com/microsoft/TypeScript/issues/37597) which suggested mismatch in version numbers, I found that we had accidentally set the version in the plugin to `5.1.28` while the rest of the repo was using `5.1.26`. 

Your `Homepage` issue I can't reproduce – lets see if the CI has the same issue.

**Comment by joshuaellis** [comment]
We're probably good here, has design QA'd this @Mcastres?

**Comment by Mcastres** [comment]
> We're probably good here, has design QA'd this @Mcastres or @annakudelska?

Thanks a lot @joshuaellis, highly appreciate this 🙏 
There was no QA for the design.

**Comment by Mcastres** [comment]
@joshuaellis do you know when this can be merged? 🙏

**Comment by Aurelsicoko** [comment]
There is a release tomorrow; it's part of the milestone, so it should be good. @joshuaellis do you confirm?

**Comment by joshuaellis** [comment]
You've got conflicts @Mcastres so you need to resolve them before you can merge the PR. It'l only dismiss my review once you've updated it.

**Comment by Mcastres** [comment]
> You've got conflicts @Mcastres so you need to resolve them before you can merge the PR. It'l only dismiss my review once you've updated it.

It concerns your changes in the `packages.json` and `yarn.lock` files. Should I accept the incoming or current changes?

**Review by joshuaellis — APPROVED** [review]

**Review by joshuaellis — APPROVED** [review]

**Pull request merged**
